onnxruntime/core/providers/qnn/qnn_execution_provider.cc

```diff
@@ -15,14 +16,28 @@
 #include "core/providers/qnn/builder/qnn_model_wrapper.h"
 #include "core/providers/qnn/builder/op_builder_factory.h"
 
-#define MEMCPY_S(dest, src, destsz, srcsz) memcpy(dest, src, std::min(destsz, srcsz))
-
 namespace onnxruntime {
 
 constexpr const char* QNN = "QNN";
 
 QNNExecutionProvider::QNNExecutionProvider(const ProviderOptions& provider_options_map)
-    : IExecutionProvider{onnxruntime::kQnnExecutionProvider, true}, runtime_options_(provider_options_map) {
+    : IExecutionProvider{onnxruntime::kQnnExecutionProvider, true},
+      runtime_options_(provider_options_map),
+      context_cache_path_("") {
+  static const std::string CONTEXT_CACHE_ENABLED = "qnn_context_cache_enable";
+  auto context_cache_enabled_pos = runtime_options_.find(CONTEXT_CACHE_ENABLED);
+  if (context_cache_enabled_pos != runtime_options_.end()) {
+    if (context_cache_enabled_pos->second == "1") {
```

Comment:
Typically, any non-zero value is considered true and only 0 is false. Although, maybe explicitly looking for `"1"` is fine too.

That said, this behavior should probably be explicitly documented in the ORT C API header file.

Reply:
right, need to update the API header file

---
In reply to: [1186430112](https://github.com/microsoft/onnxruntime/pull/15815#discussion_r1186430112) [](ancestors = 1186430112)